src/renderer/components/chart-js.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {useRef, useEffect} from 'react';
+import { Chart, ChartConfiguration, InteractionItem } from 'chart.js/auto';
+import 'chartjs-adapter-date-fns';
+
+export { InteractionItem } from 'chart.js/auto';
+interface ChartJSProps {
+    type: ChartConfiguration['type'];
+    data: ChartConfiguration['data'];
+    options: ChartConfiguration['options'];
+    plugins: ChartConfiguration['plugins'];
+    getElementAtEvent?: (items: Array<InteractionItem>, e: React.MouseEvent<HTMLCanvasElement>) => any;
+}
+const ChartJSChart = React.memo(function (props: ChartJSProps) {
+    const {data, type, options, plugins, getElementAtEvent} = props;
+
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+    const chartRef = useRef<Chart | null>(null);
+
+    useEffect(() => {
+        if (!chartRef.current) return;
+        const chart = chartRef.current;
+        const {datasets: currentDatasets} = chart.data;
+        const {datasets} = data;
+        for (const dataset of datasets) {
+            const found = currentDatasets.find((currentDataSet) => {
+                return currentDataSet.label === dataset.label;
+            });
+            if (found) {
+                Object.assign(found, dataset, {
+                    data: dataset.data
+                  });
+            } else {
+                currentDatasets.push(dataset);
+            }
+        }
+        chart.update('none');
+    }, [data]);
```

Comment:
Using a ref inside of a `useEffect` is risky business afaik.  If the ref updates the effect won't re-calculate so this is working by side-effect atm.

A cheat way of fixing this is to add `ref.current` to the dependency array but that results in delayed updates in some cases.

https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780 outlines potential options if you want to run an "effect" as the result of setting a ref 🤔